backend/core/llm_chain.py

```
@@ -3,490 +3,320 @@
-            'timestamp': time.time(),
-            'size': len(response)
-        }
-
-        # Enhanced cache cleanup with size limits
-        cleanup_cache()
-
-    except Exception as e:
-        logger.error(f"Error caching response: {e}")
+def cache_response(cache_key: str, response_chunks: List[str]):
+    """Caches the final, full response string."""
+    if not cache_key or not ENABLE_CACHING: return
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)